homeassistant/components/seventeentrack/sensor.py

```diff
@@ -111,242 +94,160 @@ async def async_setup_entry(
-                    ATTR_TRACKING_NUMBER: package.tracking_number,
-                }
-            )
-
-        self._attr_extra_state_attributes[ATTR_PACKAGES] = (
-            package_data if package_data else None
-        )
+        """Return the state of the sensor."""
+        return self.coordinator.data.summary[self._status]["quantity"]
 
-        self._state = self._data.summary.get(self._status)
+    @property
+    def extra_state_attributes(self) -> dict[str, Any] | None:
+        """Return the state attributes."""
+        return {ATTR_PACKAGES: self.coordinator.data.summary[self._status]["packages"]}
 
 
-class SeventeenTrackPackageSensor(SensorEntity):
+class SeventeenTrackPackageSensor(
+    CoordinatorEntity[SeventeenTrackCoordinator], SensorEntity
+):
     """Define an individual package sensor."""
 
     _attr_attribution = ATTRIBUTION
     _attr_icon = "mdi:package"
 
-    def __init__(self, data, package) -> None:
-        """Initialize."""
-        self._attr_extra_state_attributes = {
-            ATTR_DESTINATION_COUNTRY: package.destination_country,
-            ATTR_INFO_TEXT: package.info_text,
-            ATTR_TIMESTAMP: package.timestamp,
-            ATTR_LOCATION: package.location,
-            ATTR_ORIGIN_COUNTRY: package.origin_country,
-            ATTR_PACKAGE_TYPE: package.package_type,
-            ATTR_TRACKING_INFO_LANGUAGE: package.tracking_info_language,
-            ATTR_TRACKING_NUMBER: package.tracking_number,
-        }
-        self._data = data
-        self._friendly_name = package.friendly_name
-        self._state = package.status
-        self._tracking_number = package.tracking_number
-        self.entity_id = ENTITY_ID_TEMPLATE.format(self._tracking_number)
+    def __init__(
+        self,
+        coordinator: SeventeenTrackCoordinator,
+        tracking_number: str,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator)
+        self._attr_extra_state_attributes = {}
+        self._tracking_number = tracking_number
+        self.entity_id = ENTITY_ID_TEMPLATE.format(tracking_number)
         self._attr_unique_id = UNIQUE_ID_TEMPLATE.format(
-            data.account_id, self._tracking_number
+            coordinator.account_id, tracking_number
         )
 
     @property
     def available(self) -> bool:
         """Return whether the entity is available."""
-        return self._data.packages.get(self._tracking_number) is not None
+        package_data = self.coordinator.data.current_packages.get(
+            self._tracking_number, {}
+        )
+        return package_data.get("package") is not None
 
     @property
     def name(self) -> str:
         """Return the name."""
-        if not (name := self._friendly_name):
+        package_data = self.coordinator.data.current_packages.get(
+            self._tracking_number, {}
+        )
+        package = package_data.get("package")
+        if package is None or not (name := package.friendly_name):
```

Comment:
Afaik (I know for sure for native value and some other properties, name I am not sure), if the available property returns False, the other properties aren't requested anymore, so we shouldn't have to check again here.

(We can use square brackets instead of get

self.coordinator.data.current_packages[self._tracking_number]
)

Reply:
moved it anyway to init